browser/ui/views/side_panel/ai_chat/ai_chat_side_panel_web_view.cc

```diff
@@ -66,3 +73,35 @@ void AIChatSidePanelWebView::OnShow() {
     should_focus_ = false;
   }
 }
+
+content::WebContents* AIChatSidePanelWebView::AddNewContents(
+    content::WebContents* source,
+    std::unique_ptr<content::WebContents> new_contents,
+    const GURL& target_url,
+    WindowOpenDisposition disposition,
+    const blink::mojom::WindowFeatures& window_features,
+    bool user_gesture,
+    bool* was_blocked) {
+  auto* browser_view = BrowserView::GetBrowserViewForNativeWindow(
+      this->GetWidget()->GetNativeWindow());
+  auto* browser = browser_view->browser();
+
+  // If AI Chat is not open in the side panel, don't open the tab.
+  if (browser->browser_window_features()
+          ->side_panel_ui()
+          ->GetCurrentEntryId() != SidePanelEntryId::kChatUI) {
+    return nullptr;
+  }
+
+  // Rather than opening a new tab from the side panel we navigate the active
+  // tab next to the sidepanel.
+  auto* active_tab = browser->tab_strip_model()->GetActiveWebContents();
+  NavigateParams params(browser, target_url, ui::PAGE_TRANSITION_LINK);
+  params.source_contents = active_tab;
```

Comment:
Could you make it more explicit?


```suggestion
  // Sets source_contents and disposition so that the url can be loaded the current active tab
  params.source_contents = active_tab;
  params.disposition = WindowOpenDisposition::CURRENT_TAB;
```